topology/test/test-all.m4

```diff
@@ -85,7 +85,9 @@ PCM_DUPLEX_ADD(Passthrough, 3, 0, 0, PIPELINE_PCM_1, PIPELINE_PCM_2)
 #
 # TEST_SSP_DATA_BITS bit I2S using TEST_SSP_PHY_BITS bit sample conatiner on SSP TEST_DAI_PORT
 #
-DAI_CONFIG(TEST_DAI_TYPE, TEST_DAI_PORT, TEST_DAI_PORT, TEST_DAI_LINK_NAME,
+DAI_CONFIG(TEST_DAI_TYPE, TEST_DAI_PORT,
+	   ifelse(index(TEST_DAI_LINK_NAME, NoCodec), -1, 0 ,TEST_DAI_PORT),
```

Comment:
I think this can make everybody happy at the moment, but considering that we actually use same machine driver for normal(soft/topology/xx.m4) and test(soft/topology/test/xx) tplg files. In this case, only the first BE dai_link(with .id=0) works with test tplg file, all others doesn't work.

But I can imagine that adding an extra param for it may introduce more effort, especially for batch tplgs generated with scripts.